src/backend/backend_concept.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#ifdef __cpp_concepts
+#  include <cstddef>
+#  include "../ds/concept.h"
+#  include "../pal/pal_concept.h"
+
+namespace snmalloc
+{
+  class MetaEntry;
+
+  template<typename Meta>
```

Comment:
Please can you add doc comments to all of these explaining what the methods are supposed to do?